src/cfnlint/rules/resources/properties/AllowedPattern.py

```diff
@@ -34,18 +35,22 @@ def check_value(self, value, path, property_name, **kwargs):
         # the RegEx itself is used.
         value_pattern = kwargs.get('value_specs', {}).get('AllowedPattern', value_pattern_regex)
 
-        if value_pattern_regex:
-            regex = re.compile(value_pattern_regex)
+        if isinstance(value, (int, float)):
+            value = str(value)
```

Comment:
[seems okay to convert from looking through the existing `AllowedPatternRegex`](https://github.com/aws-cloudformation/cfn-python-lint/search?l=JSON&q=AllowedPatternRegex)

Reply:
Yea ran into an issue when this regex was looking at something was just a number.  Realized we needed to convert an int to a string to make sure the regex pattern will work.